go/libraries/doltcore/doltdb/table.go

```diff
@@ -561,6 +561,10 @@ func (t *Table) GetIndexRowData(ctx context.Context, indexName string) (durable.
 		return nil, err
 	}
 
+	if indexName == "PRIMARY" {
```

Comment:
A couple paths use this, but I think the important one was `newConstraintViolationsLoadedTable()` from `violations_fk.go`.
However, I might be able to remove this, since I added: https://github.com/dolthub/dolt/pull/4583/files#diff-665d6a79567c44e029b188e4cd624a25af54288cb1dbf621adb559597f5564adR506

Maybe using empty string instead of `PRIMARY` for the name of this index would be a good idea? It should be impossible for users to create an index with empty string name, since the name by default is the columns that make up that index.